pkg/client/unversioned/client_test.go

```diff
@@ -20,221 +20,15 @@ import (
-		// We will fix this by supporting multiple group versions in Config
-		c.ExtensionsClient = NewExtensionsOrDie(&Config{
-			Host:         c.server.URL,
-			GroupVersion: testapi.Extensions.GroupVersion(),
-		})
-	}
-	c.QueryValidator = map[string]func(string, string) bool{}
-	return c
-}
-
-func (c *testClient) Validate(t *testing.T, received runtime.Object, err error) {
-	c.ValidateCommon(t, err)
-
-	if c.Response.Body != nil && !api.Semantic.DeepDerivative(c.Response.Body, received) {
-		t.Errorf("bad response for request %#v: expected %#v, got %#v", c.Request, c.Response.Body, received)
-	}
-}
-
-func (c *testClient) ValidateRaw(t *testing.T, received []byte, err error) {
-	c.ValidateCommon(t, err)
-
-	if c.Response.Body != nil && !reflect.DeepEqual(c.Response.Body, received) {
-		t.Errorf("bad response for request %#v: expected %#v, got %#v", c.Request, c.Response.Body, received)
-	}
-}
-
-func (c *testClient) ValidateCommon(t *testing.T, err error) {
-	defer c.server.Close()
-
-	if c.Error {
-		if err == nil {
-			t.Errorf("error expected for %#v, got none", c.Request)
-		}
-		return
-	}
-	if err != nil {
-		t.Errorf("no error expected for %#v, got: %v", c.Request, err)
-	}
-
-	if c.handler.RequestReceived == nil {
-		t.Errorf("handler had an empty request, %#v", c)
-		return
-	}
-
-	requestBody := body(t, c.Request.Body, c.Request.RawBody)
-	actualQuery := c.handler.RequestReceived.URL.Query()
-	t.Logf("got query: %v", actualQuery)
-	t.Logf("path: %v", c.Request.Path)
-	// We check the query manually, so blank it out so that FakeHandler.ValidateRequest
-	// won't check it.
-	c.handler.RequestReceived.URL.RawQuery = ""
-	c.handler.ValidateRequest(t, path.Join(c.Request.Path), c.Request.Method, requestBody)
-	for key, values := range c.Request.Query {
-		validator, ok := c.QueryValidator[key]
-		if !ok {
-			switch key {
-			case unversioned.LabelSelectorQueryParam(testapi.Default.Version()):
-				validator = validateLabels
-			case unversioned.FieldSelectorQueryParam(testapi.Default.Version()):
-				validator = validateFields
-			default:
-				validator = func(a, b string) bool { return a == b }
-			}
-		}
-		observed := actualQuery.Get(key)
-		wanted := strings.Join(values, "")
-		if !validator(wanted, observed) {
-			t.Errorf("Unexpected query arg for key: %s.  Expected %s, Received %s", key, wanted, observed)
-		}
-	}
-	if c.Request.Header != "" {
-		if c.handler.RequestReceived.Header.Get(c.Request.Header) == "" {
-			t.Errorf("header %q not found in request %#v", c.Request.Header, c.handler.RequestReceived)
-		}
-	}
-
-	if expected, received := requestBody, c.handler.RequestBody; expected != nil && *expected != received {
-		t.Errorf("bad body for request %#v: expected %s, got %s", c.Request, *expected, received)
-	}
-}
-
-// buildResourcePath is a convenience function for knowing if a namespace should be in a path param or not
-func buildResourcePath(namespace, resource string) string {
```

Comment:
This function is not being used. `testapi` provides similar function. I'll just remove it.